src/MessageTrait.php

```diff
@@ -212,5 +212,14 @@ private function assertHeader($header)
         if ($header === '') {
             throw new \InvalidArgumentException('Header name can not be empty.');
         }
+
+        if (! preg_match('/^[a-zA-Z0-9\'`#$%&*+.^_|~!-]+$/', $header)) {
```

Comment:
please link to https://tools.ietf.org/html/rfc7230#section-3.2 in the `assertHeader` method phpdoc